integrations/slack/src/misc/utils.ts

```diff
@@ -140,19 +141,21 @@ const isValidUrl = (str: string) => {
 }
 
 const getOptionalProps = (ctx: IntegrationCtx, logger: IntegrationLogger) => {
-  if (ctx.configuration.botAvatarUrl) {
-    if (isValidUrl(ctx.configuration.botAvatarUrl)) {
+  const config = ctx.configuration as bp.configuration.Configuration
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❔ question:** I think the `configuration type` is inferred here. You shouldn't have to `cast` it. Is it not the case ?

Reply:
At first, I was under the impression that whatever you put in `configuration` is also available in `configurations`, so I expected the `botAvatarUrl` and `botName` options to still be available when I chose manual configuration, as opposed to oauth configuration. However, I was puzzled to see that `ctx.configuration` had the union type `bp.configuration.Configuration | bp.configurations.botToken.BotTokenConfig` so I thought I needed to cast it to `bp.configuration.Configuration`. I later realized however that the two configs were entirely separate and that I had to also define the keys from `configuration` into `configurations` if I wanted them to always be present. I updated the integration definition file accordingly, but it seems I forgot this one instance